packages/compiler/core/semantic-walker.ts

```diff
@@ -279,6 +279,20 @@ export function mapChildModels(program: Program): ReadonlyMap<ModelType, readonl
   return map;
 }
 
+const ChildModelMap = Symbol();
+/**
+ * @returns the mapping of parent model to child model.
+ * Will cache the value so it is computed only once.
+ */
+export function getChildModelMap(program: Program): ReadonlyMap<ModelType, readonly ModelType[]> {
```

Comment:
Not sure if anyone else has a better idea for caching this. I think this might cause some issue with projections. Maybe we need to have a per projection cache?